File: 백준/Silver/1181. 단어 정렬/단어 정렬.py
T = int(input())
words = []
r_words = [] #중복 제거를 위해

for i in range(T):
    word = input()
    words.append(word)

for i in words:
    if i not in r_words:
        r_words.append(i)
    
r_words.sort(key=lambda x: (len(x), x))

for i in r_words:
    print(i)

File: 백준/Silver/1181. 단어 정렬/단어 정렬.cc
#include <iostream>
#include <algorithm>
#include <string>
#include <vector>

using namespace std;

bool comp(string s1, string s2)
{
    if (s1.length() != s2.length())
    {
        return s1.length() < s2.length();
    }
    return s1 < s2;
}
int main()
{
    int N;
    vector<string> words = {};
    cin >> N;
    for (int i = 0; i < N; i++)
    {
        string tmp;
        cin >> tmp;
        words.push_back(tmp);
    }
    sort(words.begin(), words.end(), comp);
    words.erase(unique(words.begin(), words.end()), words.end());
    for (int i = 0; i < words.size(); i++)
    {
        cout << words[i] << "\n";
    }
    return 0;
}